unit-tests-java8/src/test/java/org/eclipse/collections/test/list/mutable/ArraysAsListTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.test.list.mutable;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.collections.impl.factory.Lists;
+import org.eclipse.collections.impl.test.junit.Java8Runner;
+import org.eclipse.collections.test.list.FixedSizeListTestCase;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertThrows;
+
+@RunWith(Java8Runner.class)
+public class ArraysAsListTest
+        implements FixedSizeListTestCase
+{
+    @SafeVarargs
+    @Override
+    public final <T> List<T> newWith(T... elements)
+    {
+        return Arrays.asList(elements);
+    }
```

Comment:
Here I wanted to do something similar as the ArrayListTest. I figured that Arrays.asList(...) is very similar to ArrayAdapter and I could see if the two collections pass the same tests. It turns out they do not because our fixed-size collections throw immediately if you try to remove from them, and the Arrays.asList() implementation iterates through looking for the item to remove, and only throws if it finds the item. This is a rare case where our contract deviates from the JDK and yet is obviously better. I kept the test though, with the overrides showing the different behavior.